master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -888,7 +892,10 @@ private[celeborn] class Master(
     if (authEnabled) {
       pushApplicationMetaToWorkers(requestSlots, slots)
     }
-    context.reply(RequestSlotsResponse(StatusCode.SUCCESS, slots.asInstanceOf[WorkerResource]))
+    context.reply(RequestSlotsResponse(
```

Comment:
We can introduce a flag to indicate whether the client requires a packed partition. This allows the master to determine the appropriate response instead of returning both options.